sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleTransport.scala

```diff
@@ -453,12 +459,16 @@ class DirectByteBufferPool(bufferSize: Long) extends Logging {
       new RefCountedDirectByteBuffer(ByteBuffer.allocateDirect(bufferSize.toInt), Option(this))
     } else {
       buff.clear()
+      // reset endianness to BIG_ENDIAN, as it could have changed depending on the consumer
+      // flatbuffers
+      buff.order(ByteOrder.BIG_ENDIAN)
```

Comment:
Can we add a comment why we need BIG_ENDIAN to begin with? Looks like it's the only place referencing BIG_ENDIAN byte order

Reply:
This should handle it @gerashegalov (https://github.com/NVIDIA/spark-rapids/pull/2886/commits/2efa3461313d53b343ff4dcc4f9c2d96407891b9)

Reply:
Added this: https://github.com/NVIDIA/spark-rapids/issues/2897 @gerashegalov to potentially just use LITTLE_ENDIAN consistently.